lib/manifestTools/validationRules/all/apiAccessRulesValid.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+var validationConstants = require('../../validationConstants');
+var validations = require('../../../common/validations');
+var validAccessTypes = {
+  'ios': [ 'cordova' ],
+  'windows': [ 'cordova' ],
+  'android': [ 'cordova' ],
+  'windows10': [ 'all', 'allowForWebOnly' ]
+};
+
+// Add 'none' as a valid API access type for all platforms
+for (var platform in validAccessTypes) {
+  if (validAccessTypes[platform].indexOf('none') < 0) {
+    validAccessTypes[platform].push('none');
+  }
+}
+
+module.exports = function (manifestContent, callback) {
+  var validationResults = [];
+  var apiAccessRules = manifestContent.mjs_api_access;
+
+  if (apiAccessRules && apiAccessRules instanceof Array) {

```

Comment:
You could probably avoid two levels of nesting here by doing `!` checks on the instance of Array and also on rule.platform, and returning early.